src/ray/core_worker/task_manager.h

```diff
@@ -93,13 +121,39 @@ class TaskManager : public TaskFinisherInterface {
   void OnTaskDependenciesInlined(const std::vector<ObjectID> &inlined_dependency_ids,
                                  const std::vector<ObjectID> &contained_ids) override;
 
+  /// Mark all tasks pending for a given actor as failed.
+  ///
+  /// \param[in] actor_id Actor ID.
+  void FailPendingActorTasks(const ActorID &actor_id);
+
+  /// Resubmit all tasks pending for an actor. This should be called whenever an actor
+  /// is (re)constructed.
+  ///
+  /// \param[in] actor_id Actor ID.
+  void ResubmitActorTasks(const ActorID &actor_id);
+
   /// Return the spec for a pending task.
   TaskSpecification GetTaskSpec(const TaskID &task_id) const;
 
   /// Return the number of pending tasks.
   int NumPendingTasks() const { return pending_tasks_.size(); }
 
+  /// Return the number of tasks pending for a specific actor.
+  int NumPendingActorTasks(const ActorID &actor_id) {
+    auto it = pending_actor_tasks_.find(actor_id);
+    RAY_CHECK(it != pending_actor_tasks_.end());
+    return it->second.size();
+  }
```

Comment:
Not sure if there's a policy around inline methods. Can move this to `.cc`, but it seems like a good candidate for inlining.

Reply:
We don't have a specific convention, but this seems fine. One weird thing, though, is that I think the thread safety analysis wasn't working for header files, but I could be wrong (@edoakes?).

Reply:
That's right - we should avoid having code in header files for any class with thread annotations.